akka-actor/src/main/scala/akka/util/ByteString.scala

```diff
@@ -253,9 +253,11 @@ object ByteString {
       else ByteString1(bytes, startIndex, Math.min(n, length))
 
     override def slice(from: Int, until: Int): ByteString = {
```

Comment:
the current impl (previous too) is prone to negative values, let's do a small modification to it:

```scala
      val f = Math.max(from, 0)
      if (f <= 0 && length <= until) this
      else {
        val u = Math.max(until, 0)
        if (u <= f) ByteString1.empty
        else if (0 <= f && u <= bytes.length) ByteString1(bytes, startIndex + f, until - f)
        else if (f < 0) ByteString1(bytes, startIndex, u)
        else /*(bytes.length <= u)*/ ByteString1(bytes, startIndex + f, bytes.length - f)
      }
```